dex/pair/src/locked_asset.rs

```diff
@@ -0,0 +1,146 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use crate::config;
+use crate::errors::*;
+
+#[derive(TopEncode, TopDecode, NestedEncode, NestedDecode, TypeAbi, Debug)]
+pub struct LockedAssetAttributes<M: ManagedTypeApi> {
+    pub unlock_epoch: u64,
+    pub amount: BigUint<M>,
+}
+
+#[elrond_wasm::module]
+pub trait LockedAsset: config::ConfigModule + token_send::TokenSendModule {
+    #[endpoint(setLockedAssetTokenIdFirst)]
+    fn set_locked_asset_token_id_first(&self, token_id: TokenIdentifier) {
+        self.require_permissions();
+        self.locked_asset_token_id_first().set(token_id);
+    }
+
+    #[endpoint(setLockedAssetTokenIdSecond)]
+    fn set_locked_asset_token_id_second(&self, token_id: TokenIdentifier) {
+        self.require_permissions();
+        self.locked_asset_token_id_second().set(token_id);
+    }
+
+    #[endpoint(setLockingPeriodInEpochs)]
+    fn set_locking_period_in_epochs(&self, num_epochs: u64) {
+        self.require_permissions();
+        self.locking_period_in_epochs().set(num_epochs);
+    }
+
+    #[endpoint(setLockedAssetGenerateEpochLimit)]
+    fn set_locked_asset_generate_epoch_limit(&self, epoch: u64) {
+        self.require_permissions();
+        self.locked_asset_generate_epoch_limit().set(epoch);
+    }
+
+    #[payable("*")]
+    #[endpoint(unlockAssets)]
+    fn unlock_assets(
+        &self,
+        #[payment_token] payment_token: TokenIdentifier,
+        #[payment_nonce] payment_nonce: u64,
+        #[payment_amount] payment_amount: BigUint,
```

Comment:
Since you're using NFTs for locked assets, I'd say remove this argument, since you don't need it. (amount will always be 1)

Reply:
done